realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/RealmFunctionsTest.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import androidx.test.platform.app.InstrumentationRegistry
+import io.realm.internal.util.BsonConverter
+import org.bson.*
+import org.junit.Before
+import org.junit.Test
+import java.util.stream.Collectors
+import kotlin.test.assertEquals
+
+class RealmFunctionsTest {
+
+    @Before
+    fun setup() {
+        Realm.init(InstrumentationRegistry.getInstrumentation().targetContext)
+    }
+
+    @Test
+    fun jniBsonOnlyRoundtrip() {
```

Comment:
This test is currently a bit weird. 

Normally we would test all cases through the public API, e.g. all the `BsonTest` cases would be expressed in terms of the public API.

This is a bit different though as the helper class is supposed to be used elsewhere. So, in this case, it is probably acceptable to just smoke test the public API that the serialization works.

Reply:
Yep. This is a spinout from #6810, as we suddently needed the BSON passing across multiple PRs. As we did not have a mechanism for doing JNI methods for test only, I had to put a round trip somewhere in the actual code. It will be superseeded by tests in #6810. Should of course have noted that.

Reply:
Ok 👍